File: Java/src/Balanced Binary Tree.java
import javax.swing.tree.TreeNode;

public class Solution
{
  public boolean isBalanced(TreeNode root)
  {
    if (root == null)
    {
      return true;
    }
    int left = depth(root.left);
    int right = depth(root.right);

    return Math.abs(left - right) < 2 && isBalanced(root.left) && isBalanced(root.right);

  }

  public int depth(TreeNode root)
  {
    if (root == null)
    {
      return 0;
    }
    else
    {
      return (Math.max(depth(root.left), depth(root.right)) + 1);
    }

  }
}

File: Python/Balanced Binary Tree.py
class Solution:
    # @param root, a tree node
    # @return a boolean
    def isBalanced(self, root):
        if root == None:
            return True
        left = self.depth(root.left)
        right = self.depth(root.right)
        return abs(left-right) < 2 and self.isBalanced(root.right) and self.isBalanced(root.left)

    def depth(self,root):
        if root == None:
            return 0
        return max(self.depth(root.left),self.depth(root.right)) + 1